### PR TITLE
Create a (Discord)Color using a string

### PR DESCRIPTION
**Pull request by Adrigorithm**
## Description
It allows for Colo~~u~~rs to be created from strings using constructor overloading. Please accept my first PRussy to Discord.Net :^)

## Changes
- new constructor taking a string (and an enum for possible future color formats))
- tests for this constructor

### CODE REVIEW

**Comment by Misha-133** [comment]
Since it's parsing a string value, the logic should go into a static `Parse` method
Could you do that? And also make a `bool TryParse()` one please

**Comment by Adrigorithm** [comment]
Added the bitmasking and uh the `if (rawValue.StartsWith("0x"))` was needed because people can actually create hex codes that include leading 0's :(

File: src/Discord.Net.Core/Entities/Roles/Color.cs [hunk 1]
```diff
@@ -200,6 +200,94 @@ public Color(float r, float g, float b)
         public static implicit operator uint(Color color)
             => color.RawValue;
 
+        /// <summary>
+        /// Converts the string representation of a color to a Color object.
```
Comment (Misha-133) [inline src/Discord.Net.Core/Entities/Roles/Color.cs]:
```suggestion
        ///     Converts the string representation of a color to a Color object.
```

File: src/Discord.Net.Core/Entities/Roles/Color.cs [hunk 2]
```diff
@@ -200,6 +200,94 @@ public Color(float r, float g, float b)
...
+        public static Color Parse(string rawValue, ColorType colorType = ColorType.CssHexColor)
+        {
+            if (TryParse(rawValue, out var color, colorType))
+                return color;
+
+            throw new ArgumentOutOfRangeException(nameof(rawValue), "Value must be a number of valid length - matching the specified ColorType");
+        }
+
+        /// <summary>
+        /// Converts the string representation of a color to a Color object. The return value indicates whether the conversion succeeded.
```
Comment (Misha-133) [inline src/Discord.Net.Core/Entities/Roles/Color.cs]:
```suggestion
        ///     Converts the string representation of a color to a Color object. The return value indicates whether the conversion succeeded.
```

**Review by Misha-133 — APPROVED** [review]

**Comment by itslychee** [comment]
image

**Pull request merged**